frontend/wallet/src/render/views/settings/CompressorSettings.re

```diff
@@ -0,0 +1,158 @@
+open Tc;
+
+module Styles = {
+  open Css;
+
+  let label =
+    merge([
+      Theme.Text.Body.semiBold,
+      style([color(Theme.Colors.midnight), marginBottom(`rem(0.25))]),
+    ]);
+};
+
+type modalState =
+  | Delegate(option(PublicKey.t), option(Int64.t))
+  | Stake(option(Int64.t));
+
+let intForState = state =>
+  switch (state) {
+  | Delegate(_, _) => 0
+  | Stake(_) => 1
+  };
+
+module CompressorModal = {
+  [@react.component]
+  let make = (~state, ~updateModal) =>
+    <Modal
+      title="Change consensus" onRequestClose={_ => updateModal(_ => None)}>
+      <Spacer height=0.5 />
+      <ToggleButton
+        options=[|"Delegate", "Stake"|]
+        selected={intForState(state)}
+        onChange={value =>
+          updateModal(_ =>
+            switch (value) {
+            | 0 => Some(Delegate(None, None))
+            | 1 => Some(Stake(None))
+            | _ => None
+            }
+          )
+        }
+      />
+      <Spacer height=1.5 />
+      {switch (state) {
+       | Delegate(pubkey, fee) =>
+         <>
+           <div>
+             <div className=Styles.label> {React.string("Delegate to")} </div>
+             <TextField
+               label="Key"
+               value={
+                 Option.map(~f=PublicKey.toString, pubkey)
+                 |> Option.withDefault(~default="")
+               }
+               mono=true
+               onChange={value =>
+                 updateModal(_ =>
+                   Some(Delegate(Some(PublicKey.ofStringExn(value)), fee))
+                 )
+               }
+             />
+           </div>
+           <Spacer height=1. />
+           <div>
+             <div className=Styles.label>
+               {React.string("Transaction fee")}
+             </div>
+             <TextField.Currency
+               label="Fee"
+               value={
+                 Option.map(~f=Int64.to_string, fee)
+                 |> Option.withDefault(~default="")
+               }
+               placeholder="0"
+               onChange={value => {
+                 let serializedValue =
+                   switch (value) {
+                   | "" => None
+                   | nonEmpty => Some(Int64.of_string(nonEmpty))
+                   };
+                 updateModal(_ => Some(Delegate(pubkey, serializedValue)));
+               }}
+             />
+           </div>
+         </>
+       | Stake(fee) =>
+         <>
```

Comment:
could use the same code as above, but vary the updateModal function - since the rest of the code is duplicated.

Eg. renderCurrencyInput(~f=...)